a-star.ts

```diff
@@ -87,105 +93,111 @@ namespace scene {
                     parent,
                     cost
                 );
-            } else if (!lData.visited && lData.lastCost > cost) {
+            } else if (lData.lastCost > cost) {//!lData.visited && 
                 lData.lastCost = cost;
                 lData.parent = parent;
             } else {
                 return;
             }
 
-            let h = heuristic(l);
-            // need to store extra cost on location node too, and keep that up to date
-            // if (h > parent.extraCost) {
+            const newConsideredTile = new PrioritizedLocation(
+                l,
+                cost,
+                cost + heuristic(l)
+            )
 
-            // }
 
-            consideredTiles.push(
-                new PrioritizedLocation(
-                    l,
-                    cost,
-                    h
-                )
-            );
+            if (consideredTiles.length == 0) {
+                consideredTiles.push(newConsideredTile)
+                return
+            }
+            let i = consideredTiles.length - 1
+            for (; i >= 0; i--) {  //seek&insert from end, last N are more possible hit
+                if (newConsideredTile.totalCost < consideredTiles[i].totalCost) {
+                    consideredTiles.insertAt(i + 1, newConsideredTile)
+                    return;
+                }
+            }
+            if (i < 0)
+                consideredTiles.insertAt(0, newConsideredTile)
         }
+
         updateOrFillLocation(start, null, 0);
 
-        let end: tiles.Location = null;
+        let end: SimpleLocation = null;
         while (consideredTiles.length !== 0) {
+
             const currLocation = consideredTiles.pop();
 
             if (isEnd(currLocation.loc)) {
                 end = currLocation.loc;
                 break;
             }
 
-            const row = locationRow(currLocation.loc);
-            const col = locationCol(currLocation.loc);
+            const row = currLocation.loc.row;
+            const col = currLocation.loc.col;
+
+            const neighborCost = currLocation.cost + 1000;
+            const cornerCost = currLocation.cost + 1414//1.414213562373095;

```

Comment:
```suggestion
```
move these defines after "continue" jump out, will fast a little

Reply:
I tried twice to delete this 2 lines, but after clicked the "commit suggestion" nothing happened.